src/radical/pilot/utils/misc.py

```diff
@@ -79,5 +84,152 @@ def write_dir(path):
     fout.close()
 
 
+# ------------------------------------------------------------------------------
+#
+def get_resource_configs() -> ru.Config:
+    '''
+    Return all resource configurations used by `radical.pilot`.
+
+    Returns:
+
+        :obj:`radical.utils.Config`: the resource configurations
+
+        Configurations for the individual resources are organized as sites and
+        resources:
+
+           cfgs = get_resource_configs()
+           sites = cfgs.keys()
+           for site in sites:
+               resource_names = cfgs[site].keys()
+    '''
+
+    global _rcfgs
+
+    if not _rcfgs:
+        _rcfgs = ru.Config('radical.pilot.resource', name='*', expand=False)
+
+    # return a deep copy
+    return ru.Config(cfg=_rcfgs)
```

Comment:
I would propose to use `ru.Config(from_dict=_rcfgs)`, since config is already formed, thus no need to go through all configs again

Reply:
Done, thanks